ui/src/dashboards/components/DashboardPage.tsx

```diff
@@ -31,10 +36,14 @@ interface OwnProps {
   autoRefresh: AutoRefresh
 }
 
-type Props = OwnProps & StateProps & ManualRefreshProps
+type Props = OwnProps & StateProps & ManualRefreshProps & DispatchProps
 
 @ErrorHandling
 class DashboardPage extends Component<Props> {
+  public componentWillUnmount() {
+    this.props.resetCachedQueryResults()
```

Comment:
:+1: thinking about cache invalidation

Reply:
Can't take any credit for it, it was all @ebb-tide 's idea